File: Downloader/get_downloads_count.js
const axios = require("axios");

async function getTopVersionNodeModules(nodeModuleName, numbers) {
  const countDownloadsUrl = `https://api.npmjs.org/versions/${nodeModuleName}/last-week`;

  try {
    // Response from request URL
    const response = await axios.get(countDownloadsUrl);
    const data = response.data;

    // Sort the data by values in descending order
    const sortedDataDesc = Object.fromEntries(
      Object.entries(data.downloads).sort(
        ([, valueA], [, valueB]) => valueB - valueA
      )
    );

    // Get the top {numbers} versions of the node module
    const topVersions = Object.keys(sortedDataDesc).slice(0, numbers);
    return topVersions;
  } catch (error) {
    console.error("Error fetching data:", error);
    return [];
  }
}

// Example usage
getTopVersionNodeModules("fastify", 10).then((topVersions) => {
  console.log(topVersions);
});

File: get_downloads_count.py
import requests


def get_top_version_node_modules(node_module_name, numbers):
    count_downloads_url = f'https://api.npmjs.org/versions/{node_module_name}/last-week'

    # response from request url
    response = requests.get(count_downloads_url)
    data = response.json()

    # sort the data by values in descendign order
    sorted_data_desc = dict(sorted(
        data['downloads'].items(), key=lambda x: x[1], reverse=True))

    # get the top {numbers} versions of node_module
    top_versions = list(sorted_data_desc.keys())[:numbers]
    return top_versions


get_top_version_node_modules('fastify', 10)
